lib/widgets/emoji_reaction.dart

```diff
@@ -0,0 +1,264 @@
+                  // taller than the emoji (e.g. because it needs multiple lines)
+                  child: Padding(padding: const EdgeInsets.symmetric(horizontal: 3, vertical: 1),
+                    child: Text(
+                      textWidthBasis: TextWidthBasis.longestLine,
+                      style: TextStyle(
+                        fontFamily: 'Source Sans 3',
+                        fontSize: (14 * 0.90),
+                        height: 13 / (14 * 0.90),
+                        color: labelColor,
+                      ).merge(selfVoted
+                        ? weightVariableTextStyle(context, wght: 600, wghtIfPlatformRequestsBold: 900)
+                        : weightVariableTextStyle(context)),
+                      label)),
+                ),
+              ])))));
+  }
+}
+
+class _UnicodeEmoji extends StatelessWidget {
+  const _UnicodeEmoji({
+    required this.emojiCode,
+    required this.emojiName,
+    required this.selected,
+  });
+
+  final String emojiCode;
+  final String emojiName;
+  final bool selected;
+
+  @override
+  Widget build(BuildContext context) {
+    final parsed = tryParseEmojiCodeToUnicode(emojiCode);
+    if (parsed == null) {
+      return _TextEmoji(emojiName: emojiName, selected: selected);
+    }
+    final textScaler = MediaQuery.textScalerOf(context);
+    return SizedBox(
+      width: textScaler.scale(17),
+      child: Text(
+        style: const TextStyle(fontSize: 17),
+        strutStyle: const StrutStyle(fontSize: 17, forceStrutHeight: true),
+        parsed));
+  }
+}
+
+class _ImageEmoji extends StatelessWidget {
+  const _ImageEmoji({
+    required this.emojiCode,
+    required this.emojiName,
+    required this.selected,
+  });
+
+  final String emojiCode;
+  final String emojiName;
+  final bool selected;
+
+  Widget get _textFallback => _TextEmoji(emojiName: emojiName, selected: selected);
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+
+    // Some people really dislike animated emoji.
+    final doNotAnimate = MediaQuery.of(context).disableAnimations
+      || PlatformDispatcher.instance.accessibilityFeatures.reduceMotion;
+
+    String src;
+    switch (emojiCode) {
+      case 'zulip': // the single "zulip extra emoji"
+        src = '/static/generated/emoji/images/emoji/unicode/zulip.png';
+      default:
+        final item = store.realmEmoji[emojiCode];
+        if (item == null) {
+          return _textFallback;
+        }
+        src = doNotAnimate && item.stillUrl != null ? item.stillUrl! : item.sourceUrl;
+    }
+    final parsedSrc = Uri.tryParse(src);
+    if (parsedSrc == null) {
+      return _textFallback;
+    }
+    final resolved = store.account.realmUrl.resolveUri(parsedSrc);
+
+    // Unicode emoji get scaled; it would look weird if image emoji didn't.
+    final size = MediaQuery.textScalerOf(context).scale(17);
```

Comment:
It'd be good for this 17 to be a named constant that's shared with the other 17s that are meant to equal it, in _UnicodeEmoji and _TextEmoji.